src/singletons.js

```diff
@@ -26,20 +31,26 @@ async function ERC1820Registry (funder) {
 }
 
 async function getDeployedERC1820Registry () {
-  const environment = require('./config/environment').getEnviroment();
-
-  if (environment === 'truffle') {
-    const truffleContract = require('@truffle/contract');
-    const contractAbstraction = truffleContract({ abi: ERC1820_REGISTRY_ABI });
-    contractAbstraction.setProvider(web3.currentProvider);
-
-    return contractAbstraction.at(ERC1820_REGISTRY_ADDRESS);
+  const config = getSingletonsConfig();
+  const loader = setupLoader({
```

Comment:
Have you considered using a constructor for this? `new Loader`

Reply:
I tend to avoid JS classes for reasons unknown :man_shrugging: I'm basically not sure when they are a good fit.